File: fileloader/src/main/java/app/task/fileloader/listener/FileLoaderStatusListener.java
package app.task.fileloader.listener;


import app.task.fileloader.model.FileLoaderDataModel;

public interface FileLoaderStatusListener {
    void setDone(FileLoaderDataModel fileLoaderDataModel);

    void setCancelled(FileLoaderDataModel fileLoaderDataModel);

    void onFailure(FileLoaderDataModel fileLoaderDataModel);
}

File: fileloader/src/main/java/app/task/fileloader/listener/FileLoaderStatusListener.kt
package app.task.fileloader.listener

import app.task.fileloader.model.FileLoaderDataModel

/*
 * Created by venkatesh on 10-Apr-19.
 */

interface FileLoaderStatusListener {
    fun setDone(fileLoaderDataModel: FileLoaderDataModel)

    fun setCancelled(fileLoaderDataModel: FileLoaderDataModel)

    fun onFailure(fileLoaderDataModel: FileLoaderDataModel)
}
